shell/components/form/NameNsDescription.vue

```diff
@@ -140,6 +140,18 @@ export default {
     horizontal: {
       type:    Boolean,
       default: true,
+    },
+    namespaceRules: {
```

Comment:
There's cases where we have even more inputs, maybe we pass an object with the key of the model?

```ts
rules: {
  default: () => {
    namespace: [],
    name: [],
    description: []
  },
  type: Object,  // We can convert to TS another time I guess
}
```

We should probably create a global dictionary type for these cases.
```ts
type Ruleset = (value: unknown) => string | undefined
type RulesetsComponent = Record<string, Ruleset[]>
```